### PR TITLE
fix(apigateway): remove timeout upper bound max

### LINKED ISSUE #30539: Gateway API: integration timeout cannot be over 29 seconds
### Describe the bug

AWS API Gateway timeout limit for integrations was previously 29 seconds. There has been an update to allow increasing the Gateway integration timeouts past the previous limit. https://aws.amazon.com/about-aws/whats-new/2024/06/amazon-api-gateway-integration-timeout-limit-29-seconds/

However, when implementing a Gateway API to Lambda, it seems that when synth'ing the stack, this seems to hit a validation error.

### Expected Behavior

When configuring the Gateway API type (regional/private), timeout could be set over 29 seconds.

### Current Behavior

When configuring the Gateway API, timeout cannot be over 29 seconds.

Error log when deploying/synth'ing

```
WARN AWS_SOLUTIONS_CONSTRUCTS_WARNING:  Possible override of timeout value.
Failed to synthesize: Error: Integration timeout must be between 50 milliseconds and 29 seconds.
```

### Reproduction Steps

```typescript
const ApiGateway = new ApiGatewayToLambda(
      this,
      'MyAPIGW',
      {
        lambdaFunctionProps: {
          runtime: Runtime.PROVIDED_AL2,
          handler: 'bootstrap',
          code: Code.fromAsset(join(__dirname, '..', '..', 'target', 'bin', 'linux', 'amd64', 'lambda', 'runtime')),
          timeout: Duration.minutes(5),
          environment: {
            ...
          },
        },
        apiGatewayProps: {
          defaultCorsPreflightOptions: {
            allowOrigins: ['*'],
            allowMethods: [HttpMethod.POST],
          },
          integrationOptions: {
            timeout: Duration.seconds(60),
          },
          endpointConfiguration: {
            types: [aws_apigateway.EndpointType.REGIONAL],
          },
        },
      },
    );
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### CDK CLI Version

2.145.0

### Framework Version

2.145.0

### Node.js Version

20.14.0

### OS

MacOS

### Language

TypeScript

### Language Version

_No response_

### Other information

This issue came up when using `aws-solutions-constructs`. After quick look, it seemed to point to the Gateway API core package. Original issue reference here: https://github.com/awslabs/aws-solutions-constructs/issues/1137,

### PR DESCRIPTION
**Pull request by nmussy**
### Issue # (if applicable)

Closes #30539.

### Reason for this change

AWS , see [announcement](https://aws.amazon.com/about-aws/whats-new/2024/06/amazon-api-gateway-integration-timeout-limit-29-seconds/) and [Amazon API Gateway quotas](https://docs.aws.amazon.com/apigateway/latest/developerguide/limits.html)

### Description of changes

* Updated exception of `apigateway` `Integration` to only check the `timeout` lower bound
* Added exception check of the `timeout` prop to `apigateway-v2` `WebSocketIntegration`, to match the `apigateway` v1 behavior
* Implemented the `timeout` property to `apigateway-v2` `HttpIntegration` and its sub-integrations (`HttpAlbIntegration`, `HttpLambdaIntegration`, etc.)
* Updated TSDoc

### Description of how you validated changes

Updated unit and integration tests

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-apigatewayv2/lib/http/integration.ts [hunk 1]
```diff
@@ -326,6 +345,7 @@ export abstract class HttpRouteIntegration {
         secureServerName: config.secureServerName,
         parameterMapping: config.parameterMapping,
         credentials: config.credentials,
+        timeout: config.timeout,
```
Comment (nmussy) [inline packages/aws-cdk-lib/aws-apigatewayv2/lib/http/integration.ts:348]: This ended up being a "feature" PR as well, the HTTP integration for API Gateway V2 did not implement the `timeout` prop

Comment (awsdataarchitect) [reply packages/aws-cdk-lib/aws-apigatewayv2/lib/http/integration.ts:348]: @nmussy the apigatewayv2 is for http and web sockets and the limit is not adjustable for those yet. Only for the apigateway

https://docs.aws.amazon.com/apigateway/latest/developerguide/limits.html

Revert the changes from v2

Comment (nmussy) [reply packages/aws-cdk-lib/aws-apigatewayv2/lib/http/integration.ts:342]: This change is only adding the ability to set the `timeout` prop for the API Gateway V2 HTTP integrations. I'd already added this to the WebSocket integrations previously, see #29566

Comment (awsdataarchitect) [reply packages/aws-cdk-lib/aws-apigatewayv2/lib/http/integration.ts:342]: @nmussy  thanks for making the change

**Review by scorbiere — APPROVED** [review]
LGTM, thanks for also adding the timeout property where it was missing.

**Pull request merged**